src/fides/api/ops/api/v1/endpoints/privacy_preference_endpoints.py

```diff
@@ -241,21 +295,117 @@ def save_privacy_preferences(
             detail=privacy_request_results.failed[0].message,
         )
 
-    consent_request.privacy_request_id = privacy_request_results.succeeded[0].id
-    consent_request.save(db=db)
-
+    if consent_request:
+        # If we have a verified user identity, go ahead and update the associated ConsentRequest for record keeping
+        consent_request.privacy_request_id = privacy_request_results.succeeded[0].id
+        consent_request.save(db=db)
     return upserted_current_preferences
 
 
+def verify_address(request: Request) -> None:
+    """Verify request is coming from approved address"""
+    origin = request.headers.get("Origin") or request.headers.get("Referer")
+
+    if origin:
+        parsed_url = urlparse(origin)
+        if (
+            parsed_url.scheme + "://" + parsed_url.netloc
+            not in CONFIG.security.cors_origins
+        ):
+            raise HTTPException(
+                status_code=HTTP_403_FORBIDDEN,
+                detail="Can't save privacy preferences from non-approved addresses",
+            )
+
+
+@router.get(
+    PRIVACY_PREFERENCES,
+    status_code=HTTP_200_OK,
+    response_model=Page[CurrentPrivacyPreferenceSchema],
+)
+def get_privacy_preferences_by_device_id(
+    *,
+    request: Request,
+    fides_user_device_id: str,
+    db: Session = Depends(get_db),
+    params: Params = Depends(),
+) -> AbstractPage[CurrentPrivacyPreference]:
+    """Saves privacy preferences with respect to a fides user device id.
+
+    Creates historical records for these preferences for record keeping, and also updates current preferences.
+    Creates a privacy request to propagate preferences to third party systems.
+    """
+    verify_address(request)
+    fides_user_provided_identity: Optional[
+        ProvidedIdentity
+    ] = _get_fides_user_device_id_provided_identity(
+        db=db, fides_user_device_id=fides_user_device_id
+    )
+
+    if not fides_user_provided_identity:
+        raise HTTPException(
+            status_code=HTTP_403_FORBIDDEN,
+            detail="Provided identity not found",
+        )
+
+    logger.info(
+        "Getting privacy preferences with respect to fides user provided identity"
+    )
+    query: Query[CurrentPrivacyPreference] = (
+        db.query(CurrentPrivacyPreference)
+        .filter(
+            CurrentPrivacyPreference.fides_user_device_provided_identity_id
+            == fides_user_provided_identity.id
+        )
+        .order_by(CurrentPrivacyPreference.updated_at.desc())
+    )
+    return paginate(query, params)
+
+
+@router.patch(
+    PRIVACY_PREFERENCES,
+    status_code=HTTP_200_OK,
+    response_model=List[CurrentPrivacyPreferenceSchema],
+)
+def save_privacy_preferences(
+    *,
+    request: Request,
+    db: Session = Depends(get_db),
+    data: PrivacyPreferencesCreateWithCode,
```

Comment:
For now, using a very similar request body to the existing endpoints where we save preferences with respect to a verified identity, but plan to refactor this request body to pull more items from request headers in https://github.com/ethyca/fides/issues/3133